load_confounds/parser.py

```diff
@@ -340,30 +355,51 @@ def _load_single(self, confounds_raw):
         """Load a single confounds file from fmriprep."""
         # Convert tsv file to pandas dataframe
         confounds_raw = _confounds_to_df(confounds_raw)
-
         confounds = pd.DataFrame()
+        not_found_confounds = []
 
         if "motion" in self.strategy:
-            confounds_motion = _load_motion(confounds_raw, self.motion, self.n_motion)
-            confounds = pd.concat([confounds, confounds_motion], axis=1)
+            try:
+                confounds_motion = _load_motion(confounds_raw, self.motion, self.n_motion)
+                confounds = pd.concat([confounds, confounds_motion], axis=1)
+            except ConfoundNotFoundException as exception:
+                not_found_confounds += exception.params
 
         if "high_pass" in self.strategy:
-            confounds_high_pass = _load_high_pass(confounds_raw)
-            confounds = pd.concat([confounds, confounds_high_pass], axis=1)
+            try:
+                confounds_high_pass = _load_high_pass(confounds_raw)
+                confounds = pd.concat([confounds, confounds_high_pass], axis=1)
+            except ConfoundNotFoundException as exception:
+                not_found_confounds += exception.params
 
         if "wm_csf" in self.strategy:
-            confounds_wm_csf = _load_wm_csf(confounds_raw, self.wm_csf)
-            confounds = pd.concat([confounds, confounds_wm_csf], axis=1)
+            try:
+                confounds_wm_csf = _load_wm_csf(confounds_raw, self.wm_csf)
+                confounds = pd.concat([confounds, confounds_wm_csf], axis=1)
+            except ConfoundNotFoundException as exception:
+                not_found_confounds += exception.params
 
         if "global" in self.strategy:
-            confounds_global_signal = _load_global(confounds_raw, self.global_signal)
-            confounds = pd.concat([confounds, confounds_global_signal], axis=1)
+            try:
+                confounds_global_signal = _load_global(confounds_raw, self.global_signal)
+                confounds = pd.concat([confounds, confounds_global_signal], axis=1)
+            except ConfoundNotFoundException as exception:
+                not_found_confounds += exception.params
 
         if "compcor" in self.strategy:
-            confounds_compcor = _load_compcor(
-                confounds_raw, self.compcor, self.n_compcor
+            try :
```

Comment:
Getting all those `try` `except` argument in the code is not elegant. As they are all written around private functions, maybe we could simply add decorators. Something like: 
```
def trace_error(func):
    def wrapper_trace_error(*args, **kwargs):
        try:
            func(*args, **kwargs)
        except ConfoundNotFoundException as exception:
            not_found_confounds += exception.params
        return func(*args, **kwargs)
    return wrapper_trace_error
``` 
Then all you would need to do is add this decorator before all of the `_load_xxx` functions. 
```
@trace_error
def _load_global( ...
``` 
See https://realpython.com/primer-on-python-decorators/ 
Then the code of `_load_single` would be unchanged, except for the last statement `if not_found_confounds:` wich could be turned into a short helper function.